### PR TITLE
NewMinedBlock: Mention the Miner

### PR DESCRIPTION
**Pull request by fosskers**
This makes it easy to see in our logging who was credited for mining each block.

### CODE REVIEW

File: src/Chainweb/Miner/Coordinator.hs [hunk 1]
```diff
@@ -142,8 +145,9 @@ publish lf (MiningState ms) cdb bh = do
     c <- _cut cdb
     let !phash = _blockPayloadHash bh
     res <- runExceptT $ do
-        T2 p pl <- HM.lookup phash ms ?? "BlockHeader given with no associated Payload"
-        c' <- tryMonotonicCutExtension c bh !? "Newly mined block for outdate cut"
+        T3 m p pl <- HM.lookup phash ms ?? "BlockHeader given with no associated Payload"
+        let !miner = m ^. position @1 . _Unwrapped
```
Comment (mightybyte) [inline src/Chainweb/Miner/Coordinator.hs]: Ditch `DataKinds` and `Data.Generics.Product.Positions` and just use `view _1`.

Comment (fosskers) [reply src/Chainweb/Miner/Coordinator.hs]: I don't think those are the same thing. `position @1` grabs the first field of any product type with a `Generic` instance.

Comment (mightybyte) [reply src/Chainweb/Miner/Coordinator.hs]: Oh, I saw the `T3` and incorrectly thought it was referring to the tuple.  In that case we should definitely not use this overcomplicated product stuff.  Add accessors to the `Miner` type.  That's dramatically more understandable.

Comment (mightybyte) [reply src/Chainweb/Miner/Coordinator.hs]: Better yet, just use the `minerId` lens that is already there.

Comment (fosskers) [reply src/Chainweb/Miner/Coordinator.hs]: Good catch, I somehow overlooked the `minerId` lens.

File: src/Chainweb/Miner/Coordinator.hs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE AllowAmbiguousTypes #-}
 {-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
```
Comment (mightybyte) [inline src/Chainweb/Miner/Coordinator.hs]: Do we still need this?

**Pull request merged**